File: src/test/java/de/tanschmi/aoc2024/dec1/Dec1Test.java
package de.tanschmi.aoc2024.dec1;

import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;

import static org.junit.jupiter.api.Assertions.*;

class Dec1Test {
    Dec1 task = new Dec1();

    @Test
    void step1() throws IOException {
        final long expected = 1603498L;
        File f = new File(ClassLoader.getSystemResource("inputs/dec1.txt").getFile());
        long result = task.step1(f);
        assertEquals(expected, result);
    }

    @Test
    void step2() throws IOException {
        final long expected = 25574739L;
        File f = new File(ClassLoader.getSystemResource("inputs/dec1.txt").getFile());
        long result = task.step2(f);
        assertEquals(expected, result);
    }

    @Test
    void testStep1() {
    }

    @Test
    void testStep2() {
    }
}

File: src/test/kotlin/de/tanschmi/kotlin/aoc2024/dec1/Dec1Test.kt
package de.tanschmi.kotlin.aoc2024.dec1

import org.junit.jupiter.api.Test
import java.io.File
import kotlin.test.assertEquals

class Dec1Test{
    var dec1 = Dec1()

    @Test
    fun testDec1(){
        var expected: Long = 1603498

        var input = File(ClassLoader.getSystemResource("inputs/dec1.txt").file)
        var sum: Long = dec1.step1(input)
        assertEquals(expected, sum)
    }

    @Test
    fun testDec2() {
        var expected: Long = 25574739

        var input = File(ClassLoader.getSystemResource("inputs/dec1.txt").file)
        var sum = dec1.step2(input)
        assertEquals(expected, sum)
    }
}